WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupGoogleFragment.java

```diff
@@ -0,0 +1,194 @@
+                                showErrorDialog(getString(R.string.login_error_generic));
+                                break;
+                        }
+                    }
+                } else if (result == RESULT_CANCELED) {
+                    AnalyticsTracker.track(AnalyticsTracker.Stat.LOGIN_SOCIAL_BUTTON_FAILURE);
+                    AppLog.e(T.NUX, "Google Signup Failed: result was CANCELED.");
+                } else {
+                    AnalyticsTracker.track(AnalyticsTracker.Stat.LOGIN_SOCIAL_BUTTON_FAILURE);
+                    AppLog.e(T.NUX, "Google Signup Failed: result was not OK or CANCELED.");
+                    showErrorDialog(getString(R.string.login_error_generic));
+                }
+
+                break;
+        }
+    }
+
+    @Override
+    protected void showAccountDialog() {
+        Intent signInIntent = Auth.GoogleSignInApi.getSignInIntent(mGoogleApiClient);
+        startActivityForResult(signInIntent, REQUEST_SIGNUP);
+    }
+
+    // Remove scale from photo URL path string.  Current URL matches /s96-c, which returns a 96 x 96
+    // pixel image.  Removing /s96-c from the string returns a 512 x 512 pixel image.  Using regular
+    // expressions may help if the photo URL scale value in the returned path changes.
+    private String removeScaleFromGooglePhotoUrl(String photoUrl) {
+        Pattern pattern = Pattern.compile("(/s[0-9]+-c)");
+        Matcher matcher = pattern.matcher(photoUrl);
+        return matcher.find() ? photoUrl.replace(matcher.group(1), "") : photoUrl;
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onAuthenticationChanged(AccountStore.OnAuthenticationChanged event) {
+        if (event.isError()) {
+            AppLog.e(T.API, "SignupGoogleFragment.onAuthenticationChanged: " + event.error.type + " - " + event.error.message);
+        // Continue with signup since account was created.
+        } else if (event.createdAccount) {
+            mGoogleListener.onGoogleSignupFinished(mDisplayName, mGoogleEmail, mPhotoUrl);
+        // Continue with login since existing account was selected.
+        } else {
+            mLoginListener.loggedInViaSocialAccount(mOldSitesIds, true);
+        }
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onSocialChanged(AccountStore.OnSocialChanged event) {
+        if (event.isError()) {
+            AppLog.e(T.API, "SignupGoogleFragment.onSocialChanged: " + event.error.type + " - " + event.error.message);
+
+            switch (event.error.type) {
+                // WordPress account exists with input email address, and two-factor authentication is required.
+                case TWO_STEP_ENABLED:
+                    AnalyticsTracker.track(AnalyticsTracker.Stat.SIGNUP_SOCIAL_2FA_NEEDED);
+                    ToastUtils.showToast(getContext(), getString(R.string.signup_user_exists, mGoogleEmail),
```

Comment:
Can we shorten the length of `signup_user_exists`? Even with a long toast, it's not easy to read the whole thing before it disappears, especially when you're not expecting it.

Also, the phrase "continue with login" is a bit cryptic and/or unfriendly. Might want to ask Editorial for better copy.

Reply:
Yep, I totally agree.  I meant to add the `[Status] Needs Copy Review` label for wordsmithing help from Editorial.  I'll add that and `[Status] Needs Design Review`.